scripts/flail.py

```diff
@@ -12,60 +12,79 @@
 
 class Flail:
     """
-        The flaing consists in converting a file to a C header containing all the contents
-        as an uint array
+        The input file can be either an object file (e.g. a goto binary file)
+        or a list of C source file. The flailng converts the content of an input file
+        into an uint array. The resulting array will be written in a C header.
```

Comment:
This comment is misleading: The input to this script can be file(s) of any type. There is no interpretation happening, "list of C source file" and "goto binary file" are misleading.

What `flail.py` does (in a quite obscure way via external tools, Python could actually do it itself via [`ord`](https://docs.python.org/3/library/functions.html#ord) - maybe slower, don't know) is to generate either (depending on a command-line option) a single C source file (and possibly also a header for it) containing one symbol for each input file defining the binary content of the file (as `char` array) and one symbol defining its length (as `unsigned int`), or one corresponding C source file (possibly including header) for each input file.

Reply:
The `flail.py` is just a translation of an older script (`flail.sh`). I am actually trying to make it python only as one of the dependencies keeps causing problems in the Windows Build.

> a single C source file (and possibly also a header for it) containing one symbol for each input file defining the binary content of the file (as char array)

You mean accepting a list of files as inputs and then generating an amalgamated version?

Reply:
> The `flail.py` is just a translation of an older script (`flail.sh`).

I suspected that much. It's good to get rid of using these external tools.

> You mean accepting a list of files as inputs and then generating an amalgamated version?

Yes, I added support for multiple inputs some time back in cb442ce6d14a6f3da08cd7165a0be5a6bdf3f8ba. It's used to bundle the c2goto/{library,header} files for use in <https://github.com/esbmc/esbmc/blob/005ad337a051760ce22ddaaa618620b3794e13c7/src/c2goto/cprover_library.cpp#L151>